process/headerCheck/headerSignatureVerify.go

```diff
@@ -278,18 +285,26 @@ func (hsv *HeaderSigVerifier) getLeader(header data.HeaderHandler) (crypto.Publi
 	return hsv.keyGen.PublicKeyFromByteArray(leaderPubKeyValidator.PubKey())
 }
 
-func (hsv *HeaderSigVerifier) copyHeaderWithoutSig(header data.HeaderHandler) data.HeaderHandler {
+func (hsv *HeaderSigVerifier) copyHeaderWithoutSig(header data.HeaderHandler) (data.HeaderHandler, error) {
 	headerCopy := header.ShallowClone()
-	headerCopy.SetSignature(nil)
-	headerCopy.SetPubKeysBitmap(nil)
-	headerCopy.SetLeaderSignature(nil)
+	err := headerCopy.SetSignature(nil)
+	if err != nil {
+		return nil, err
+	}
+
+	err = headerCopy.SetPubKeysBitmap(nil)
+	if err != nil {
+		return nil, err
+	}
+
+	err = headerCopy.SetLeaderSignature(nil)
 
```

Comment:
if err != nil {
    return nil, err
}

and L302: return headerCopy, nil

Otherwise this method could return headerCopy and err != nil

Reply:
done